build.gradle

```diff
@@ -0,0 +1,57 @@
+plugins {
+    id 'java-library'
+    id "com.diffplug.gradle.spotless" version "3.27.0"
+    id 'maven-publish'
+    id 'signing'
+    id 'io.codearte.nexus-staging' version '0.21.2'
+}
+
+group 'com.adscore'
+version '1.0.4' //version of samples match with the version of the client-libs-java project
+
+sourceCompatibility = 1.8
+ 
+repositories {
+    mavenLocal()
+    mavenCentral()
+}
+
+dependencies {
+    compile 'com.adscore:adscore-client-libraries:1.0.4'
+
+    implementation 'com.github.seancfoley:ipaddress:5.2.1'
```

Comment:
do we need this dep?

Reply:
Well, as long as `adscore-client-libraries` is not a fat jar the answer is yes.

But I think we should make it a fat jar. I don`t know why it`s not it yet, but It does not sound good to force users of our library to provide required dependencies. At least it`s not a common practice for libraries.

I think making the lib as a fat jar, with full required dependencies (so only seancfoley:ipaddress for now) could be done as a separate commit for version 1.0.4.

Reply:
The library should not be a fat jar (https://stackoverflow.com/questions/50179525/for-official-maven-repositories-do-people-release-fat-jars). All required dependencies should be resolved by maven/gradle. I guess the issue here might be the fact that you are using id 'java-library'

Reply:
Okay, the library should not be a fat jar, thanks for pointing to an answer that explains that.

But sample must be a fat jar, otherwise, the user is not able to execute it.

e.g 
If
`gradle jar` only packs sample classes,
then
`java -jar build/libs/client-libs-java-samples-1.0.4.jar base-sample` 

won`t work. end up with:
```
Exception in thread "main" java.lang.NoClassDefFoundError: com/adscore/signature/SignatureVerifier
```

I suggest not invest time into it. Works with compile + fatJar. When Gradle version 7.0.0 comes, we'll take care of switching from `compile` into `implementation`.

Reply:
As written above, the problem with building a fat-jar is fixed.

The answer to the initial question: `do we need this dep?` is yes ;)

Reply:
well I've removed it from sample myself and wasn't needed ?:)

Reply:
Oh, indeed it works. IDK how I was building it before... 
Thanks for checking that and correcting me!